crates/bevy_ecs/src/component.rs

```diff
@@ -556,6 +598,7 @@ pub struct ComponentInfo {
     descriptor: ComponentDescriptor,
     hooks: ComponentHooks,
     required_components: RequiredComponents,
+    required_by: Vec<ComponentId>,
```

Comment:
Hmm, I think I'll change this to a `HashSet`. It's more correct semantically, and it avoids duplicate values when e.g. you first require a component indirectly and later directly, causing it to be added to this list twice. I could also add guards against adding duplicates, but I think a `HashSet` is simpler.

Reply:
Big fan!